torch/jit/__init__.py

```diff
@@ -35,6 +35,13 @@
 if sys.version_info[0] > 2:
     import pathlib
 
+def _get_type_attr(type, aname):
```

Comment:
I'd rather have the 'dim' and scalarType methods handle returning None rather than throw/catch an exception. It makes things hard to debug (`catch throw` now catches spurious things), and it can mask a real exception that is not the one that is expected.